typeid/typeid-go/typeid_test.go

```diff
@@ -0,0 +1,136 @@
+package typeid
+
+import (
+	"fmt"
+	"testing"
+)
+
+func ExampleNew() {
+	tid := Must(New("prefix"))
+	fmt.Printf("New typeid: %s\n", tid)
+}
+
+func ExampleNew_withoutPrefix() {
+	tid := Must(New(""))
+	fmt.Printf("New typeid without prefix: %s\n", tid)
+}
+
+func ExampleFromString() {
+	tid := Must(FromString("prefix_00041061050r3gg28a1c60t3gf"))
+	fmt.Printf("Prefix: %s\nSuffix: %s\n", tid.Type(), tid.suffix)
+	// Output:
+	// Prefix: prefix
+	// Suffix: 00041061050r3gg28a1c60t3gf
+}
```

Comment:
Why these? Could we wrap them in a test that checks the returned values are as expected?

Reply:
No, these are examples, which go treats differently so that they can be rendered as part of the documentation. If I turn them into tests, they will no longer appear as examples in the doc.